force-app/main/default/classes/DMLOperations.cls

```diff
@@ -97,10 +101,12 @@ public with sharing class DMLOperations {
     public static void updateOpportunityStage(Id oppId, String newStage) {
         // Retrieve the Opportunity from the database
         // [SELECT Id FROM Opportunity WHERE Id = :oppId]
+        Opportunity oppToUpdate = [SELECT Id FROM Opportunity WHERE Id = :oppId];
```

Comment:
It's pretty advanced but try created an opportunity in memory, setting the id and the stage and doing the update. You don't need to SOQL if you do this.